server/game/gameSystems/PlayCardSystem.ts

```diff
@@ -33,15 +40,37 @@ export class PlayCardSystem<TContext extends AbilityContext = AbilityContext> ex
         ignoredRequirements: [],
         optional: false,
         entersReady: false,
-        playType: PlayType.PlayFromHand
+        playType: PlayType.PlayFromHand,
+        nested: false
     };
 
     public eventHandler(event, additionalProperties): void {
         const player = event.player;
         const newContext = (event.playCardAbility as PlayCardAction).createContext(player);
+
+        /* if (this.properties.costAdjusterProperties) {
+            this.queueApplyCostAdjusterGameSteps(this.properties.costAdjusterProperties, newContext);
+        }*/
         event.context.game.queueStep(new AbilityResolver(event.context.game, newContext, event.optional));
     }
 
+    private queueApplyCostAdjusterGameSteps(costAdjusterProperties: ICostAdjusterProperties, context: AbilityContext) {
+        const applyCostAdjusterSystem = new PlayerLastingEffectSystem({
+            duration: Duration.UntilEndOfAction,
+            effect: modifyCost(Object.assign(costAdjusterProperties, { limit: AbilityLimit.perGame(1), playingTypes: context.playType }))
+        });
+        const effectEvents = [];
+        applyCostAdjusterSystem.queueGenerateEventGameSteps(effectEvents, context, {});
+        context.game.queueSimpleStep(() => context.game.openEventWindow(effectEvents), 'open event window for application of cost adjuster');
+    }
+
+    private makeCostAdjuster(properties: ICostAdjusterProperties, context: AbilityContext) {
+        if (properties) {
```

Comment:
If null is an acceptable value of properties then we should add `| null` to the type for that parameter